php-cs-fix 3.23.0, phpstan 1.10.29

pmattmann opened the pull request:
closes #3710

pmattmann committed:
php-cs-fix 3.23.0, phpstan 1.10.29

pmattmann committed:
update php-cs-fixer, phpstan

pmattmann committed:
fix AssertBelongsToSameCamp

usu reviewed (commented):
- Did you find out, where the changes were coming from? It seems related to [phpdoc_add_missing_param_annotation](https://cs.symfony.com/doc/rules/phpdoc/phpdoc_add_missing_param_annotation.html), but I didn't see any changes on this related to php-cs-fixer 3.23.0
- I would prefer proper PHP type declarations instead of only type annotations (where possible)
- Many of the mixed types could actually be typed properly (string, array, etc.). I would prefer that we fix this already know (at least the easy ones). Once this is merged, it becomes more difficult to differentiate the real "mixed" vs. the lazy "mixed" ones

pmattmann commented:
> * Many of the mixed types could actually be typed properly (string, array, etc.). I would prefer that we fix this already know (at least the easy ones). Once this is merged, it becomes more difficult to differentiate the real "mixed" vs. the lazy "mixed" ones

I agree - let me try

pmattmann committed:
use concrete types

usu committed:
use type declaration instead of annotation

usu committed:
merge devel

usu committed:
fix typo

BacLuc committed:
revert CreateColumnLayoutTest of use type declaration instead of annotation

This reverts the change of CreateColumnLayoutTest of commit 0a3d499,
because then we would need to change the interface of all child classes.
Psalm, phpstan and php-cs-fixer don't complain, we can do that later.
